### ISSUE TITLE
projectile-switch-project-action is ignored

### ISSUE THREAD

**Issue by catap** [hlissner/doom-emacs#4357]
**What did you expect to happen?**

Switch to project as it defined

**What actually happened?**

Uses default value.

**Additional details:**

I put into `~/.doom.d/config.el` and when I call anywhere else `(projectile-switch-project-by-name "~/src/doom-emacs")` it switches me to `magit-status` inside this project.

That is that I'd like to have.

Anyway, when I call `C-c p p` and select `~/src/doom-emacs` it switched via default way (helm?):

![Screenshot 2020-12-03 at 00 44 41](https://user-images.githubusercontent.com/37775/100945036-c5383780-3500-11eb-8d04-0e9ba1e7246a.png)

**Steps to reproduce:**
1. put to end of `~/.doom.d/config.el` select a desire way to switch like `(setq projectile-switch-project-action 'projectile-vc)`.
2. `C-c p p` and open any project with VCS.
3. Switch by default way to switch

**System information:**
<details><pre>
```
SYSTEM  type       darwin
        config     x86_64-apple-darwin20.1.0
        shell      /opt/local/bin/zsh
        uname      Darwin 20.1.0 Darwin Kernel Version 20.1.0: Sat Oct 31 00:07:11 PDT 2020; root:xnu-7195.50.7~2/RELEASE_X86_64 x86_64
        path       (/opt/local/bin /opt/local/sbin ~/.cargo/bin /usr/local/bin /usr/bin /bin /usr/sbin /sbin /Applications/VMware Fusion.app/Contents/Public /Library/TeX/texbin /usr/local/MacGPG2/bin /Library/Apple/usr/bin /Applications/MacPorts/Emacs.app/Contents/MacOS ~/bin ~/src/doom-emacs/bin /Applications/MacPorts/Emacs.app/Contents/MacOS/libexec/)
EMACS   dir        ~/src/doom-emacs/
        version    28.0.50
        build      Dec 02, 2020
        buildopts  --prefix=/opt/local --disable-silent-rules --without-dbus --without-gconf --without-libotf --without-m17n-flt --with-gmp --with-gnutls --with-json --with-xml2 --with-modules --infodir /opt/local/share/info/emacs --with-ns --with-lcms2 --without-harfbuzz --without-imagemagick --without-xaw3d --with-nativecomp --with-rsvg --with-xwidgets 'CFLAGS=-pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.0.sdk -arch x86_64' 'CPPFLAGS=-I/opt/local/include -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.0.sdk' 'LDFLAGS=-L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-no_pie -Wl,-syslibroot,/Library/Developer/CommandLineTools/SDKs/MacOSX11.0.sdk -arch x86_64'
        features   JPEG TIFF GIF PNG RSVG GLIB NOTIFY KQUEUE ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS XIM NS MODULES NATIVE_COMP THREADS XWIDGETS JSON PDUMPER LCMS2
        traits     (gui server-running chemacs envvar-file)
DOOM    dir        ~/.doom.d/
        version    2.0.9
        font       #<font-object -*-Menlo-normal-normal-normal-*-12-*-*-*-m-0-iso10646-1>
        theme      doom-one
        build      HEAD -> develop, origin/develop f622d5171 2020-12-01 18:09:43 -0500
        elc-files  0
        modules    (:completion company ivy :ui doom doom-dashboard doom-quit (emoji +unicode) fill-column hl-todo indent-guides modeline ophints (popup +defaults) treemacs unicode vc-gutter vi-tilde-fringe :editor file-templates fold snippets :emacs dired electric undo vc :checkers syntax (spell +aspell +flyspell +everywhere) grammar :tools (eval +overlay) lookup magit :os macos :lang (cc +lsp) emacs-lisp (latex +latexmk +cdlatex +fold +lsp) lua markdown sh :config (default +bindings +smartparens))
        packages   (n/a)
        unpin      (n/a)
        elpa       (n/a)
```
</pre></details>

**Comment by hlissner** [comment]
It's being ignored because you are actually triggering `counsel-projectile-switch-project` _not_ `projectile-switch-project`. The counsel version uses `counsel-projectile-switch-project-action` instead, which the workspace module wraps to add integration with workspaces.

TL;DR, change `+workspaces-switch-project-function`, not `projectile-switch-project-action`.

**Comment by catap** [comment]
@hlissner thanks. I never thought that function with label `projectile-switch-project` is calling `counsel-projectile-switch-project`.

But it makes sense and revert my trust in our world.

Thanks!

**Comment by catap** [comment]
@hlissner I haven't use `workspaces` (you can see it from details) and `+workspaces-switch-project-function` doesn't help.

(yes, I run `doom sync` and restarted emacs to be double sure)

![Screenshot 2020-12-03 at 00 52 58](https://user-images.githubusercontent.com/37775/100945667-eea59300-3501-11eb-9e94-b3acc6efd667.png)

**Comment by hlissner** [comment]
If you don't use workspaces, change `counsel-projectile-switch-project-action` instead.
